src/pfe/file-watcher/server/test/functional-test/suites/project/tests/project-event.ts

```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+import mocha from "mocha";
+import { expect } from "chai";
+import * as _ from "lodash";
+import path from "path";
+
+import { ProjectCreation, updateProjectForNewChange } from "../../../lib/project";
+import { SocketIO } from "../../../lib/socket-io";
+
+import * as eventConfigs from "../../../configs/event.config";
+import * as project_configs from "../../../configs/project.config";
+import * as timeoutConfigs from "../../../configs/timeout.config";
+import { fail } from "assert";
+
+export function projectEventTest(socket: SocketIO, projData: ProjectCreation, projectLang: string): void {
+    describe("updateProjectForNewChange function", () => {
+        afterEach("clear socket events", () => {
+            socket.clearEvents();
+        });
+
+        const chunk = Math.random() * 1000;
+        const data: any = {
+            "projectID": projData.projectID,
+            "timestamp": Date.now(),
+            "chunk": chunk,
+            "chunk_total": chunk + Math.random() * 100,
+            "eventArray": []
+        };
+
+        const requiredKeys = ["projectID", "timestamp", "chunk", "chunk_total", "eventArray"];
+        for (const key of requiredKeys) {
+            it(`call updateProjectForNewChange without required key: ${key}`, async() => {
+                const testData = _.cloneDeep(data);
+                delete testData[key];
+                const info: any = await updateProjectForNewChange(testData.projectID, testData.timestamp, testData.chunk, testData.chunk_total, testData.eventArray);
+                expect(info);
+                expect(info.statusCode);
+                expect(info.statusCode).to.equal(400);
+                expect(info.error);
+                expect(info.error).to.haveOwnProperty("msg");
+                expect(info.error.msg).to.equal("Bad request. projectID, timestamp, chunk, chunk_total and eventArray are required.");
+            });
+        }
+
+        const types = ["MODIFY", "CREATE"];
+
+        for (const type of types) {
+            describe(`update file with type = ${type.toLowerCase()}`, () => {
+                updateProject(type);
+            });
+        }
+
+        function updateProject(type: string): void {
+            let filesList = ["file1", "file2", "file3"];
```

Comment:
For testing `CREATE` event, why would we need to test it three times for `file1`, `file2` and `file3` instead of just test one time?

Reply:
Sure, we can only do one file.